rules/integrations/o365/credential_access_antra_id_device_reg_via_oauth_redirection.toml

```diff
@@ -0,0 +1,97 @@
+[metadata]
+creation_date = "2025/04/30"
+integration = ["o365"]
+maturity = "production"
+updated_date = "2025/04/30"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies attempts to register a new device in Microsoft Entra ID after OAuth authentication with authorization code grant. Adversaries may use OAuth phishing techniques to obtain an OAuth authorization code, which can then be exchanged for access and refresh tokens. This rule detects a sequence of events where a user principal authenticates via OAuth, followed by a device registration event, indicating potential misuse of the OAuth flow to establish persistence or access resources.
+"""
+from = "now-9m"
+index = ["filebeat-*", "logs-o365.audit-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Microsoft 365 OAuth Redirect to Device Registration for User Principal"
+references =[
+    "https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth2-auth-code-flow",
+    "https://www.volexity.com/blog/2025/04/22/phishing-for-codes-russian-threat-actors-target-microsoft-365-oauth-workflows/"
+]
+risk_score = 73
+rule_id = "fcd2e4be-6ec4-482f-9222-6245367cd738"
+severity = "high"
+tags = [
+    "Domain: Cloud",
+    "Domain: SaaS",
+    "Data Source: Microsoft 365",
+    "Data Source: Microsoft 365 Audit Logs",
+    "Use Case: Identity and Access Audit",
+    "Tactic: Credential Access",
+    "Resources: Investigation Guide",
+]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+sequence by related.user with maxspan=30m
+[authentication where event.action == "UserLoggedIn" and
+ o365.audit.ExtendedProperties.RequestType == "OAuth2:Authorize" and o365.audit.ExtendedProperties.ResultStatusDetail == "Redirect" and
+ o365.audit.UserType: ("0", "2", "3", "10")] // victim source.ip
+[authentication where event.action == "UserLoggedIn" and
+ o365.audit.ExtendedProperties.RequestType == "OAuth2:Token" and o365.audit.ExtendedProperties.ResultStatusDetail == "Success"] // attacker source.ip to convert oauth code to token
+[web where event.dataset == "o365.audit" and event.action == "Add registered users to device."] // user.name is captured in related.user
+'''
+note = """## Triage and analysis
```

Comment:
Just throw note after name for sequence. Then maybe add newlines for investigation guides. Rest looks good!